polaris-service/src/main/java/org/apache/polaris/service/catalog/IcebergCatalogAdapter.java

```diff
@@ -171,28 +173,39 @@ public Response updateProperties(
         .build();
   }
 
+  private EnumSet<AccessDelegationMode> parseAccessDelegationModes(String accessDelegationMode) {
+    EnumSet<AccessDelegationMode> delegationModes =
+        AccessDelegationMode.fromProtocolValuesList(accessDelegationMode);
+    if (!delegationModes.isEmpty() && !delegationModes.contains(VENDED_CREDENTIALS)) {
+      throw new IllegalArgumentException(
+          "Unsupported access delegation mode: " + accessDelegationMode);
+    }
```

Comment:
Minor suggestion: using `Preconditions.checkArgument()`?

Reply:
Also I think we shouldn't throw in case of empty, as the callers will do something differently for empty `delegationModes`. 
https://github.com/polaris-catalog/polaris/blob/main/polaris-service/src/main/java/org/apache/polaris/service/catalog/IcebergCatalogAdapter.java#L197-L197

Reply:
updated